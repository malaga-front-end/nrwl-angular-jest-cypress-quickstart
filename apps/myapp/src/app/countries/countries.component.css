ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  li {
    cursor: pointer;
    font: 200 20px/1.5 Helvetica, Verdana, sans-serif;
    border-bottom: 1px solid #ccc;
    text-decoration: none;
    color: #000;
    display: block;
    width: 300px;
  
    -webkit-transition: font-size 0.3s ease, background-color 0.3s ease;
    -moz-transition: font-size 0.3s ease, background-color 0.3s ease;
    -o-transition: font-size 0.3s ease, background-color 0.3s ease;
    -ms-transition: font-size 0.3s ease, background-color 0.3s ease;
    transition: font-size 0.3s ease, background-color 0.3s ease;
  }
  
  li:hover {
    font-size: 30px;
    background: #f6f6f6;
  }
  