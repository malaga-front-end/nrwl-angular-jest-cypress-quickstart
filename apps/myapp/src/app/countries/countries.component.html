<ul>
    <li *ngFor="let country of countries$ | async" (click)="sendCity(country.capital)">{{ country.name }}</li>
</ul>