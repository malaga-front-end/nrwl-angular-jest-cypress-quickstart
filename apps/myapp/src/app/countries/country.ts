export interface Country {
    name: string;
    capital: string;
}