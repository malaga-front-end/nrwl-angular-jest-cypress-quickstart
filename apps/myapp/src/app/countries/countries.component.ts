import { Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs';
import { SharedService } from '../shared.service';
import { CountriesService } from './countries.service';
import { Country } from './country';

@Component({
  selector: 'myapp-countries',
  templateUrl: './countries.component.html',
  styleUrls: ['./countries.component.css']
})
export class CountriesComponent implements OnInit {

  countries$!: Observable<Country[]>;

  constructor(
    private countriesService: CountriesService,
    private sharedService: SharedService
  ) { }

  ngOnInit(): void {
    this.countries$ = this.countriesService.getCountries();
  }

  sendCity(city: string) {
    this.sharedService.sendCity(city);
  }
}
