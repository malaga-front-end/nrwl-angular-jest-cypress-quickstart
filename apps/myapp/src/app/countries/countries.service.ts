import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Country } from './country';

@Injectable({
  providedIn: 'root'
})
export class CountriesService {

  private readonly url = 'http://localhost:3000/countries';


  constructor(private httpClient: HttpClient) { }

  public getCountries(): Observable<Country[]> {
    return this.httpClient.get<Country[]>(this.url);
  }
}
