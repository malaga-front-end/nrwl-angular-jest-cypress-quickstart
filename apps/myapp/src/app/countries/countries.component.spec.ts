import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';

import { CountriesComponent } from './countries.component';
import { CountriesService } from './countries.service';
import { SharedService } from '../shared.service';
import { of } from 'rxjs';
import { By } from '@angular/platform-browser';

jest.mock('./countries.service');
jest.mock('../shared.service');

describe('CountriesComponent', () => {
  let component: CountriesComponent;
  let fixture: ComponentFixture<CountriesComponent>;
  let countriesService: CountriesService;
  let sharedService: SharedService;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [ CountriesComponent ],
      providers: [ CountriesService, SharedService ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    countriesService = TestBed.inject(CountriesService);
    sharedService = TestBed.inject(SharedService);

    jest.spyOn(countriesService, 'getCountries').mockReturnValue(of([
      { name: 'Spain', capital: 'Madrid' },
      { name: 'France', capital: 'Paris'}
    ]));

    jest.spyOn(sharedService, 'sendCity');

    fixture = TestBed.createComponent(CountriesComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should call CountriesService.getCountries()', () => {
    expect(countriesService.getCountries).toHaveBeenCalled();
  });

  it('should send capital city after clicking on a country', () => {
    const firstElement = fixture.debugElement.queryAll(By.css('li'))[0].nativeElement;
    firstElement.click();
    expect(sharedService.sendCity).toHaveBeenCalledWith('Madrid');
  });

  it('should render a list of countries', () => {
    expect(fixture).toMatchSnapshot();
  });
});
