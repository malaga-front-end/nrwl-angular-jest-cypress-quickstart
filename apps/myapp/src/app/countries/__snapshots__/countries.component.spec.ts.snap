// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesComponent should render a list of countries 1`] = `
<myapp-countries
  __ngContext__={[Function LRootView]}
  countries$={[Function Observable]}
  countriesService={[Function CountriesService]}
  sharedService={[Function SharedService]}
>
  <ul>
    <li>
      Spain
    </li>
    <li>
      France
    </li>
  </ul>
</myapp-countries>
`;
